packages/core/strapi/lib/services/server/policy.js

```diff
@@ -11,11 +11,22 @@ const resolvePolicies = route => {
   const { pluginName, apiName } = route.info || {};
   const policiesConfig = getPoliciesConfig(route);
 
-  const policies = policiesConfig.map(policyName => {
-    return policy.get(policyName, { pluginName, apiName });
-  });
+  const policiesMiddleware = async (ctx, next) => {
+    const context = policy.createPolicyContext('koa', ctx);
 
-  return [...policies, bodyPolicy];
+    for (const policyName of policiesConfig) {
+      const resolvedPolicy = await policy.get(policyName, { pluginName, apiName });
+      const result = await resolvedPolicy({ ctx: context, strapi });
+
+      if (!result) {
```

Comment:
In term of convenience what do you think about check if result is === false instead so that if you don't return (a.k.a return undefined) it will pass ?

Reply:
That's a good point!

Reply:
Do you think it might be safer to do something like `if (result === true || result === undefined)` instead. That way we only accept true and "no returns" as valid options. Everything else will be consider as false.

Reply:
LGTM yes 👍